### PR TITLE
Make autolinks case insensitive

### PR DESCRIPTION
**Pull request by UziTech**
**Marked version:** master

**Markdown flavor:** Markdown.pl|CommonMark|GitHub Flavored Markdown|n/a

## Description

Make urls case insensitive

implements https://github.com/markedjs/marked/pull/1350#issuecomment-427164249

```js
marked("hTtP://fOo.CoM");
// <p><a href=\"hTtP://fOo.CoM\">hTtP://fOo.CoM</a></p>
```

- closes #1350

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression

## Committer

In most cases, this should be a different person than the contributor.

- [x] Draft GitHub release notes have been updated.
- [x] CI is green (no forced merge required).
- [x] Merge PR

### CODE REVIEW

File: test/specs/marked/marked.json [hunk 1]
```diff
@@ -83,6 +83,18 @@
     "html": "<p>``<em>test`</em></p>",
     "example": 14
   },
+  {
+    "section": "Autolinks",
+    "markdown": "HTTP://FOO.COM",
+    "html": "<p><a href=\"HTTP://FOO.COM\">HTTP://FOO.COM</a></p>",
+    "example": 15
```
Comment (styfle) [inline test/specs/marked/marked.json]:
```suggestion
    "example": 17
```

File: test/specs/marked/marked.json [hunk 2]
```diff
@@ -83,6 +83,18 @@
...
+  },
+  {
+    "section": "Autolinks",
+    "markdown": "hTtP://fOo.CoM",
+    "html": "<p><a href=\"hTtP://fOo.CoM\">hTtP://fOo.CoM</a></p>",
+    "example": 16
```
Comment (styfle) [inline test/specs/marked/marked.json]:
```suggestion
    "example": 18
```

File: test/specs/marked/marked.json [hunk 3]
```diff
@@ -83,6 +83,18 @@
     "html": "<p>``<em>test`</em></p>",
     "example": 14
   },
+  {
+    "section": "Autolinks",
```
Comment (styfle) [inline test/specs/marked/marked.json]: Is it a problem if the example numbers collide?

We just had example 15 added in master: https://github.com/markedjs/marked/blob/master/test/specs/marked/marked.json

**Comment by styfle** [comment]
/cc @roygold7 since PR 1350 was started but didn't finish.

I want to make sure Roy is aware of this PR and that the reason for the new PR is no movement on the old PR.

I don't want to end up in [this scenario](https://twitter.com/timocov_/status/1065501522073665536).

File: test/specs/marked/marked.json (hunk 3 above)
Comment (UziTech) [reply test/specs/marked/marked.json]: I think the example numbers are just used in the shouldPassButFail array and the test descriptions.

The `marked` tests shouldn't ever fail so maybe we should replace `example` with `description` so they won't collide as often

**Comment by roygold7** [comment]
@styfle I appreciate the comment but I'd be just as happy if the issue was resolved in a different PR. I'm not looking for credit.

File: test/specs/marked/marked.json (hunk 3 above)
Comment (styfle) [reply test/specs/marked/marked.json]: > maybe we should replace example with description so they won't collide as often

I'm fine with that. Let's save that change for a different PR though. This looks good now.

**Comment by davisjam** [comment]
LGTM. This should not have a ReDoS impact.

**Pull request merged**